### PR TITLE
[MD]Fix data source link to point to correct page, and use in page redirect instead of reloading

### PR DESCRIPTION
**Pull request by zhongnansu**
### Description

[MD]Fix data source link to point to correct page, and use in page redirect instead of reloading

### Issues Resolved

#6581 
#6582 

## Screenshot
iShot_2024-04-22_23 45 41

## Testing the changes

1. create a data source and create index pattern with this data source
2. go to index pattern page, and click the data source name link under column "Data source connection", it should redirect to the corresponding data source detail page, without triggering a page reload.
3. Test the change works for dropdown header, and no data source component, since global var `DSM_APP_ID` is being used at those places. See below gif

iShot_2024-04-23_15 57 54

## Changelog

- fix: Fix data source link to point to correct page, and use in page redirect instead of reloading

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data_source_management/public/components/data_source_column/data_source_column.tsx [hunk 1]
```diff
@@ -4,18 +4,15 @@
...
+import { IndexPatternTableColumn } from '../../../../index_pattern_management/public';
+import { getApplication, getDataSources } from '../utils';
 import { DataSourceTableItem } from '../../types';
+import { DSM_APP_ID } from '../../plugin';
 
-type DataSourceColumnItem = DataSourceTableItem & { relativeUrl: string };
-type DataSourceMap = Map<string, DataSourceColumnItem> | undefined;
+type DataSourceMap = Map<string, DataSourceTableItem> | undefined;
 
 export class DataSourceColumn implements IndexPatternTableColumn<DataSourceMap> {
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_column/data_source_column.tsx:17]: do we have test for this class?

File: src/plugins/data_source_management/public/plugin.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ export interface DataSourceManagementPluginStart {
   getAuthenticationMethodRegistry: () => IAuthenticationMethodRegistry;
 }
 
-export const DSM_APP_ID = 'dataSourceManagement';
+export const DSM_APP_ID = 'dataSources';
```
Comment (BionIT) [inline src/plugins/data_source_management/public/plugin.ts:44]: Looks like we are changing the global constant which is used in multiple places, can we tests if all the places using the constant are working as expected?

Comment (zhongnansu) [reply src/plugins/data_source_management/public/plugin.ts:44]: yes, it's also used in dropdown header, and no data source component. Added video to the PR description -> testing section

File: src/plugins/data_source_management/public/components/data_source_column/data_source_column.tsx (hunk 1 above)
Comment (zhongnansu) [reply src/plugins/data_source_management/public/components/data_source_column/data_source_column.tsx:17]: Currently we don't, let me add it

**Comment by zhongnansu** [comment]
Failed CI are not related to the change of this PR

**Review by bandinib-amzn — APPROVED** [review]

**Pull request merged**